buildSrc/src/main/groovy/org/elasticsearch/gradle/DependenciesInfoTask.groovy

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.artifacts.Dependency
+import org.gradle.api.artifacts.DependencySet
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+
+
+/**
+ * A task to gather information about the dependencies and export them into csv file.
+ *
+ * The following information are gathered:
+ * <ul>
+ *     <li>name: name that identifies the library (groupId:artifactId)</li>
+ *     <li>version</li>
+ *     <li>URL: link to have more information about the dependency.</li>
+ *     <li>license: <a href="https://spdx.org/licenses/">SPDX license</a> identifier, custom license or UNKNOWN.</li>
+ * </ul>
+ *
+ */
+public class DependenciesInfoTask extends DefaultTask {
+
+    /** Dependencies to gather information from. */
+    @Input
+    public DependencySet dependencies
+
+    /** Directory to read license files */
+    @InputDirectory
+    public File licensesDir = new File(project.projectDir, 'licenses')
+
+    @OutputFile
+    File outputFile = new File(project.buildDir, "reports/dependencies/dependencies.csv")
+
+    public DependenciesInfoTask() {
+        description = 'Create a CSV file with dependencies information.'
+    }
+
+    @TaskAction
+    public void generateDependenciesInfo() {
+        final StringBuilder output = new StringBuilder()
+
+        for (Dependency dependency : dependencies) {
+            // Only external dependencies are checked
+            if (dependency.group != null && dependency.group.contains("elasticsearch") == false) {
+                output.append(dependency.group).append(':').append(dependency.name).append(',')
```

Comment:
I think it would be easier to "see" the format if using a single append with string substitutions:

```
String url = createURL(dep.group, dep.name, dep.version)
String licenseType = getLicenseType(dep.group, dep.name)
output.append("${dep.group}:${dep.name},${dep.version},${url},${licenseType}\n")
```

Reply:
You're right, updated.